src/rezup/container.py

```diff
@@ -414,36 +429,108 @@ def use(self, run_script=None):
 
         else:
             # Launch subprocess
+            replacement = dict()
+            environment = self._compose_env()
+            shell_name, shell_exec = self._get_shell()
 
-            env = self._compose_env()
-            env["PATH"] = os.pathsep.join([
-                str(self._path / "bin"),
-                env["PATH"]
-            ])
-            # use `pythonfinder` package if need to exclude python from PATH
+            if run_script and os.path.isfile(run_script):
+                # run shell script and exit
+                block = False
+                replacement["__REZUP_DO__"] = "script"
+                replacement["__REZUP_DO_SCRIPT__"] = os.path.abspath(run_script)
 
-            if run_script:
+            elif run_script:
+                # run shell command and exit
                 block = False
-                if os.path.isfile(run_script):
-                    env["__REZUP_SCRIPT__"] = os.path.abspath(run_script)
-                else:
-                    env["__REZUP_COMMAND__"] = run_script
+                replacement["__REZUP_DO__"] = "command"
+                replacement["__REZUP_DO_COMMAND__"] = run_script
+
             else:
+                # interactive shell
                 block = True
-
-            shell_name, _ = shell.get_current_shell()
-            cmd = shell.get_launch_cmd(shell_name, self._path, block=block)
-
-            popen = subprocess.Popen(cmd, env=env)
+                replacement["__REZUP_SHELL__"] = shell_exec
+                prompt = "rezup (%s) " % self._container.name()
+                prompt = shell.format_prompt_code(prompt, shell_name)
+                environment.update({
+                    "REZUP_PROMPT": os.getenv("REZUP_PROMPT", prompt),
+                })
+
+            launch_script = shell.generate_launch_script(
+                shell_name,
+                dst_dir=tempfile.mkdtemp(),
+                replacement=replacement
+            )
+            cmd = shell.get_launch_cmd(shell_name,
+                                       shell_exec,
+                                       launch_script,
+                                       block=block)
+
+            popen = subprocess.Popen(cmd, env=environment)
```

Comment:
The shell launch scripts are not all getting installed in each revision any more. Only one script that matched the shell (login shell) will get generated into a temp dir per use.